src/pages/api/v1/applications/[id].ts

```diff
@@ -71,17 +73,91 @@ const deleteHandler = async (req: NextApiRequest, res: NextApiResponse) => {
       res.status(400).json({
         message: `Invalid Input: 'id' was not a number`,
       });
+=======
+const getApplication = async (id: number, res: NextApiResponse) => {
+  try {
+    const data = await prisma.application.findUnique({
+      where: { id },
+    });
+
+    if (!data) {
+      return res
+        .status(404)
+        .json({ message: `Could not find Application with id ${id}` });
+>>>>>>> a06d29925004ad0c24f40e5ccb2e08b503843228
     }
+
+    return res.status(200).json(data);
+  } catch (error) {
+    return res.status(400).json({
+      message: `Invalid Input: 'id' was not a number`,
+    });
   }
 };
 
+<<<<<<< HEAD
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'GET') {
     getApplication(req, res);
   } else if (req.method === 'PATCH') {
     updateHandler(req, res);
   } else if (req.method === 'DELETE') {
     deleteHandler(req, res);
+=======
+type UpdateApplicationsRequest = Prisma.ApplicationUpdateInput;
+
+const updateHandler = async (
+  id: number,
+  input: UpdateApplicationsRequest,
+  res: NextApiResponse
+) => {
+  try {
+    const data = await prisma.application.update({
+      where: { id },
+      data: input,
+    });
+
+    return res.status(200).json(data);
+  } catch (error) {
+    res.status(404).json({
+      message: `Could not find Application with id ${id}`,
+    });
+  }
+};
+
+const deleteHandler = async (id: number, res: NextApiResponse) => {
+  try {
+    const data = await prisma.application.delete({
+      where: { id },
+    });
+
+    return res.status(200).json(data);
+  } catch (error) {
+    return res.status(404).json({
+      message: `Could not find Application with id ${id}`,
+    });
+  }
+};
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { id: rawId } = req.query as { id: string };
+
+  const id = parseInt(rawId);
+
+  if (Number.isNaN(id)) {
+    return res.status(400).json({
+      message: `Invalid Input: 'id' was not a number`,
+    });
+  }
+
+  if (req.method === "GET") {
+    getApplication(id, res);
+  } else if (req.method === "PATCH") {
+    const { status } = req.body;
+    updateHandler(id, { status }, res);
+  } else if (req.method === "DELETE") {
+    deleteHandler(id, res);
+>>>>>>> a06d29925004ad0c24f40e5ccb2e08b503843228
```

Comment:
Looks like there is a merge conflict marker, due to my changes clashing with your changes.

That's my bad for not telling you to pull the changes onto your local machine/branch.

I guess we'll both keep an eye out for this next time.

I'll look after this for you :smile:.